### ISSUE TITLE
No sftp support?

### ISSUE THREAD

**Issue by devhell** [rustic-rs/rustic#389]
Hi!

Love the project, thank you so much! When I tried to use my existing restic repository (Hetzner Storagebox), rustic lets me know that the sftp backend is not supported. Am I missing something? :)

Thank you :D

**Comment by aawsome** [comment]
You are right, sftp is so far only supported via the `rclone` backend. In fact, so far only local and rclone backend are implemented. 
I do hesitate a bit to implement more backends as this also may introduce maintenance burden, but would favor more to work on #74 if more backends are desirable.
That said, if anyone is up with a PR to support sftp or other backend (maybe within a [feature](https://doc.rust-lang.org/cargo/reference/features.html) ) - I would happily accept it!

**Comment by aawsome** [comment]
Some info if anyone wants to implement a sftp backend:
- I think the crate to use is  ssh2, see https://docs.rs/ssh2/latest/ssh2/struct.Sftp.html
- One needs to implement the `ReadBackend` and `WriteBackend` traits, see e.g. `src/backends/local.rs` and `src/backends/rest.rs`
- I'm not sure about the best strategy for retrying on errors, but `rest.rs` can be a good example how to handle these retries.
- Finally, the backend given in the command line is chosen in the `ChooseBackend`, see `src/backends/choose.rs`. Currently this is an Enum struct which itself needs to implement `ReadBackend` and `WriteBackend` - so, just some dumb copy&paste needed here. (Maybe in future I'll change this to hold a Trait Object, that will simplify some code here..)

**Comment by devhell** [comment]
Ah, I didn't realize that it was only available via the rclone backend. Thank you for the feedback! I'm afraid I'm in no position to help with the implementation. But I'll keep my eye on your project as I'd like to help with documentation. :beers:

**Comment by aawsome** [comment]
Help with documentation is also highly appreciated! Feel free to open any PR which adds or improves the docu!

**Comment by devhell** [comment]
Just to add to this, I've now tried `rustic` on my `restic` repo on Hetzner's storagebox via `rclone`. The result is, I'm guessing, down to `rclone`:

```
$ rustic repoinfo
[WARN] Error error decoding response body: Odd number of digits at line 1 column 482760 at 367.252424ms, retrying
```
It will continue backing-off forever, it seems (I stopped it once the retry would be attempted after 82 seconds).

PS: I did `repoinfo` because I wanted to see what `rustic` would make of it before trying any other actions.

**Comment by aawsome** [comment]
Sorry for the late reply. There must be some additional files in your repository. Valid files are all named by an sha256 hex (i.e. a 64 digit hexadecimal number). #415 should fix this as it ignores all additional files.

**Comment by devhell** [comment]
Awesome, thanks!

**Comment by aawsome** [comment]
> It will continue backing-off forever, it seems (I stopped it once the retry would be attempted after 82 seconds).

I just checked this: It should retry for a maximum of 10 minutes and then finally fail...

**Comment by tbm** [comment]
I just discovered rustic. I'd also like sftp support natively (but also can't contribute code).
